app/controllers/tests_controller.rb

```diff
@@ -1,10 +1,30 @@
 class TestsController < ApplicationController
-  before_action :check_is_logged_in
+  before_action :check_is_logged_in, only: [:index, :new, :create]
+  before_action :check_is_admin_permission, only: [:new, :create]
+
+  def new
+    @test = Test.new
+  end
+
+  def create
+    @test = Test.new test_params
+    if @test.save
+      flash[:success] = t ".success_create"
+      redirect_to root_path
+    else
+      render "new"
+    end
+  end
 
   def index
```

Comment:
should order methods follow: index > show > new > create > edit > update > destroy